contracts/Marketplace.sol

```diff
@@ -1,51 +1,324 @@
+  - Code beghind onlyOwner:
+  modifier onlyOwner() {
+        require(owner() == _msgSender(), "Ownable: caller is not the owner");
+        _;
+    }
+ */
+
+// * Counters Utility
+/** A simple way to get a counter that can only be incremented or decremented
+  - Provides counters that can only be incremented or decremented by one.
+  - Functions that includes:
+    - current() => Take the current value of the counter.
+    - increment() => Increment the counter by one.
+    - decrement() => Decrement the counter by one.
+ */
+
+// * ReentrancyGuard
+/** ReentrancyGuard is used to prevent reentrancy attacks.
+  - It has one modifier => nonReentrant() - We can use this in a widraw() function.
+  - And one function => constructor()
+  */
+
+// * Uses of the smart contract [SELLER]
+/**
+  - Approve an NFT to market contract
+  - Create a market item with listing fee
+  - Wait for a buyer to buy the NFT
+  - Receive the price value of the NFT
+   */
+
+// * Uses of the smart contract [BUYER]
+/**
+  - Buy an NFT from the market contract paying the price value
+  */
+
+// * Purchase process
+/**
+  - Transfer the price value to the seller
+  - Transfer the NFT from the seller to the buyer
+  - Transfer the listing fee to the market owner
+  - Change marketItemState from Created to Release
+   */
+
+// ! We will wait until the item is sold to transfer the NFT from the seller to the buyer
+
+// * Handle Errors (If I'm not wrong, this saves gas when showing errors and not using Strings for it)
+// Example of error with parameters => error InsufficientBalance(uint256 available, uint256 required);
+/** Using them on the function:
+ revert InsufficientBalance({
+      available: balance[msg.sender],
+      required: amount
+  });
+ */
+error PriceMoreThanOneWei();
+error FeeEqualToListingFee();
+error IdLessOrEqualItemCount();
+error ItemMustBeOnMarketplace();
+error SubmitActualPrice();
+
+contract Marketplace is ReentrancyGuard, Ownable {
+    // * EXTERNAL LIBRARIES START >>>
+    // Create 2 variables with Counters to increment or decrement them by 1
+    using Counters for Counters.Counter;
+    Counters.Counter private _itemCounter; // Start from 1 => Items created on the marketplace
+    Counters.Counter private _itemSoldCounter; // => Items already solt in the marketplace
+
+    // Just in case we need to do maths with our values/uint256 ...
+    using SafeMath for uint256;
+
+    // ? Can we implement this ?
+    // There is a function some lines below that uses the address library to check if the address is a contract.
+    using Address for address;
+    // * EXTERNAL LIBRARIES END >>>
+
+    // * Variables/Structs/Enums declaration START >>>
+    // The market has an owner which is the contract deployer. Listing fee will be going to market owner when an NFT item is sold in the market.
+    uint256 public listingFee = 0.025 ether;
+
+    // ?  Variable to know who is the owner of the MarketItem ?
+    address payable public marketOwner;
```

Comment:
I believe this is intended to be the address that will receive market fees. You would not use a global non-container variable to store the owner of individual MarketItems since there can be a lot of them.

This doesn't look to be needed though: this contract is an `Ownable` which means it already has an "owner" set to the deployer's address. Since this contract's constructor also sets `marketOwner` to the deployer address (`msg.sender` in the constructor), they're the same anyway.